refactor: add minimal fims-demo vignette (#772)

RicardoYin01 opened the pull request:
<!---
Thanks for opening a PR. This commented text will **NOT** appear in the final PR. Toggle between Write and Preview to see what your PR will look like without the comments.
-->

# What is the feature?
* Added a minimal FIMS model running example vignette
* Renamed the original fims-demo.Rmd to a detailed version (fims-demo-detailed.Rmd)

# How have you implemented the solution?
* Created a new vignette (fims-demo.Rmd) with only the minimal necessary steps as required
* Kept only the core steps needed to run a FIMS model: setup, data, parameters, initialize and fit
* Removed all non-essential content, such as update_parameters() calls and result visualization sections
* Renamed the original detailed version

# Does the PR impact any other area of the project, maybe another repo?
* No

# Related Issue:
Closes #772

kellijohnson-NOAA reviewed (changes requested):
Thank you for your Pull Request to FIMS. Unfortunately, we require all PRs are to the dev branch rather than to the main branch, which is really annoying given that GitHub suggests that forks only copy the main branch. Can you please rebase your branch to origin dev? If you need help doing this, I am more than happy to help and there are multiple ways to go about doing this. Second, here are some more detailed comments.
1. I ❤️ the new naming scheme of the files, 👍 thank you.
2. You will need to [list the new vignette in the documentation here](https://github.com/NOAA-FIMS/FIMS/blob/582bbdea356faf95fb760ad166d1f06325d7b41f/pkgdown/_pkgdown.yml#L21) so it will be linked in the website
4. The call to `library(ggplot2)` can be removed rom line 22 given that we are no longer plotting anything.

This is a non-blocking comment and just for your awareness so please do not feel you need to change this. I personally would have moved the old file using `git mv fims-demo.Rmd fims-demo-detailed.Rmd` to keep the history of that file associated with it and then made a new file called `fims-demo.Rmd` after that but this doesn't really matter.

Thank you for this and taking the initiative to do a pull request without being asked. We love outside contributions and wish you the best of luck 🍀 on getting into the Google Summer of Code for 2025.

RicardoYin01 commented:
Hi, Kelli. Thanks for the review! I've rebased my branch onto dev and force-pushed the updated commits. Then I updated the `fims-demo-detailed.Rmd` with the changes in the dev branch, listed the both vignette in `_pkgdown.yml`, and removed the call to `library(ggplot2)`, as you requested. Please let me know if there's anything else I should adjust!

kellijohnson-NOAA commented:
@RicardoYin01 thank you for your contributions to FIMS, I am closing this PR but please do not be alarmed, your changes have been brought into dev via 0e240dab80e3d71aa44de0c197f47d17427544d1 and will be merged into main upon the next release. I merged them in myself to avoid some conflicts with some reason branches that have also been merged in. Thank you again for your efforts.
